File: src/main/java/com/jss/osiris/modules/miscellaneous/service/SalesComplainOriginService.java
package com.jss.osiris.modules.miscellaneous.service;

import java.util.List;

import com.jss.osiris.modules.miscellaneous.model.SalesComplainOrigin;

public interface SalesComplainOriginService {

        public List<SalesComplainOrigin> getComplainOrigins();

        public SalesComplainOrigin addOrUpdateComplainOrigin(SalesComplainOrigin salesComplainOrigin);
}

File: client/src/app/modules/tiers/services/sales.complain.origin.service.ts
import { HttpClient, HttpParams } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { AppRestService } from 'src/app/services/appRest.service';
import { SalesComplainOrigin } from '../../miscellaneous/model/SalesComplainOrigin';

@Injectable({
  providedIn: 'root'
})
export class SalesComplainOriginService extends AppRestService<SalesComplainOrigin>{

  constructor(http: HttpClient) {
    super(http, "tiers");
  }

  getComplainOrigins() {
    return this.getListCached(new HttpParams(), "sales-complain-origins");
  }

   addOrUpdateComplainOrigin(salesComplainOrigin: SalesComplainOrigin) {
    return this.addOrUpdate(new HttpParams(), "sales-complain-origin", salesComplainOrigin, "Enregistré", "Erreur lors de l'enregistrement");
  }
}
